### PR TITLE
Add custom pre and post processor functions

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add custom pre and post processor functions and a Bedrock connector documentation

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/pretrained-models.md [hunk 1]
```diff
@@ -51,4 +51,5 @@ The following table provides a list of sentence transformer models and artifact
...
+| `huggingface/sentence-transformers/paraphrase-mpnet-base-v2` | 768-dimensional dense vector space. | Yes | - [model_url](https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/paraphrase-mpnet-base-v2/1.0.0/torch_script/sentence-transformers_paraphrase-mpnet-base-v2-1.0.0-torch_script.zip)<br>- [config_url](https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/paraphrase-mpnet-base-v2/1.0.0/torch_script/config.json) | - [model_url](https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/paraphrase-mpnet-base-v2/1.0.0/onnx/sentence-transformers_paraphrase-mpnet-base-v2-1.0.0-onnx.zip)<br>- [config_url](https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/paraphrase-mpnet-base-v2/1.0.0/onnx/config.json) |
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/pretrained-models.md:55]: This model version is `1.0.0`,  which is different with other models. Suggest add a new `version` column to show the model version

File: _ml-commons-plugin/extensibility/connectors.md [hunk 2]
```diff
@@ -192,17 +192,17 @@ The `parameters` section requires the following options when using `aws_sigv4` a
...
-The following example request creates a standalone Cohere connection:
+The following example request creates a standalone Cohere connector:
 
 ```json
 POST /_plugins/_ml/connectors/_create
 {
-  "name": "YOUR CONNECTOR NAME",
-  "description": "YOUR CONNECTOR DESCRIPTION",
-  "version": "YOUR CONNECTOR VERSION",
+  "name": "<YOUR CONNECTOR NAME>",
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/extensibility/connectors.md:200]: Suggest add pre/post process functions https://github.com/opensearch-project/ml-commons/blob/main/docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md

File: _ml-commons-plugin/extensibility/blueprints.md [hunk 3]
```diff
@@ -173,6 +173,43 @@ The remote text embedding model output must be a two-dimensional float array, ea
 ]
 ```
 
+## Custom pre- and post-processing functions
+
+You can write your own pre- and post-processing functions specifically for your model format. For example, the following Amazon Bedrock connector definition contains custom pre- and post-processing functions for the Amazon Bedrock Titan model:
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/extensibility/blueprints.md:178]:
```suggestion
You can write your own pre- and post-processing functions specifically for your model format. For example, the following Amazon Bedrock connector definition contains custom pre- and post-processing functions for the Amazon Bedrock Titan embedding model:
```

File: _ml-commons-plugin/pretrained-models.md (hunk 1 above)
Comment (dhrubo-os) [reply _ml-commons-plugin/pretrained-models.md:55]: @kolchfa-aws  Could you please add these models too mentioned in this issue: https://github.com/opensearch-project/opensearch-py-ml/issues/334?

Comment (kolchfa-aws) [reply _ml-commons-plugin/pretrained-models.md:55]: @dhrubo-os We've added the first 3 already (see [this section](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sparse-encoding-models)). For the other two, could you provide the dimensionality and autotruncation so I can fill the table? Thanks!

**Comment by kolchfa-aws** [comment]
@ylwu-amzn I implemented the changes. Please review again to make sure everything looks good. Thanks!

File: _ml-commons-plugin/pretrained-models.md (hunk 1 above)
Comment (dhrubo-os) [reply _ml-commons-plugin/pretrained-models.md]: @kolchfa-aws `distiluse-base-multilingual-cased-v1 ` --> 512

`paraphrase-mpnet-base-v2` --> 768

Auto truncation is yes for both of them.

Comment (kolchfa-aws) [reply _ml-commons-plugin/pretrained-models.md]: Thanks! Added.

**Review by vagimeli — APPROVED** [review]
@kolchfa-aws Great writing to add, correct, and clarify information!

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**